### DISCUSSION TITLE
Is the grammar section intentionally empty?

### DISCUSSION THREAD

**Discussion by Ethan-Arrowood** [micromark/micromark#48] [Q&A] [answered]
I've been digging around the whole micromark ecosystem and discovered the [Grammar](https://github.com/micromark/micromark#grammar) section of these docs. Is it intentionally empty like that? Am I just missing something obvious and have too much tunnel vision right now?

**Comment by ChristianMurphy** [comment, marked as answer]
It isn't empty, the context free grammar for markdown is:

```
markdown = .*
```

with `.*` meaning all characters can be in a valid markdown program/document.
Which is also explained by the sentence above the grammar which links to http://trevorjim.com/a-specification-for-markdown/

**Reply by Ethan-Arrowood** [reply]
ohhh i get it. I was expecting something like

```
Document ::= Block
Block ::= Heading | Paragraph | (etc.)
Heading ::= '#' {1..6} + Identifier
```

But yeah that post sorta sums it up. Okay cool; I was in fact missing something 😂

**Reply by wooorm** [reply]
I added it yesterday and it’s indeed intentional like this. But maybe it could use some more explanation!

**Reply by Ethan-Arrowood** [reply]
There may be some value in adding the counter point. Something like:

You may be expecting a BNF form grammar syntax such as:
```
X ::= Y + Z
```
However, Markdown is not like other programming languages; in fact it's not really a programming language at all. [Insert points made by previous articles and reference back].

**Reply by wooorm** [reply]
I added something a couple hours ago: https://github.com/micromark/micromark#grammar. Does that cover sort of what you’re going for?

**Reply by wooorm** [reply]
I wanted to skip the “what is or is not a programming language” that the original article makes: it’s hotly debated, deciding whether something is programming or not based on whether there are error is a bit weird in my opinion, and it isn’t really important here anyway 🤷‍♂️

**Reply by Ethan-Arrowood** [reply]
your adjustment is perfect. Thank you for adding it!

And yes true - best to avoid that topic 😅
